Single Level sonic disk with arduno pro mini

richardFirth opened the pull request:
hey,

I put the files I used to create my single-leveled version of the sonic disk in a separate folder in my fork.
I also put a processing3 sketch i used for testing in there, with a slightly modified version of your test arduino master code.
I figured it makes more sense to merge with yours rather then my having it in a fork only.

platisd commented on RF_WRK/README.md:
```diff
@@ -0,0 +1,11 @@
+# Richard Version
+
+*I wasn't sure how to add this to the inital repo, so I put my stuff in this folder so as not to mess up however your stuff is organized.*
```
I think this is the most important thing we should determine before continuing with this review. From your perspective, is this an addition/enhancement of the original project or a new project?

I am asking, since if it is the latter then it deserves to be in its own repository. Otherwise we'll need to do some changes in the structure and documentation in order to make it consistent with the rest. Let me know what you think so we can move this forward! :+1:

platisd replied:
Just checked out the schematics. Looks very cool, nice idea and execution! :smiley:

However, I think that this is a _unique_ project that can and can and should stand on its own as it follows a different concept (with multiple boards), Arduino footprint, processing sketch that seems to be serving a specific purpose and so on.

This is a fascinating project and I'd feel bad for overshadowing it by stuffing it in a folder of this repo. :) Please let me know your thoughts on this.

richardFirth commented:
I had thought of it as an implementation of your sonic disc, but I can also see how it's a separate thing. The term 'fork' is very appropriate for this case actually.

 I came across the article about this thing on Hackaday a while back, and then when I had completed my omnibot I thought it would be a cool thing to add to it. 

I checked out your repo and was able to see what you had done very quickly since it was cleanly documented. (I also didn't know how to I2C slave arduino's at the time so I picked that up from your code as well)

I can see how blending in an additional variant of the PCB & assembly could add confusion to someone who is trying to understand the initial thing, so it makes sense to keep the alternate schematics and fabrication files separate, since they aren't a strict addition to the original.

For the processing3 test script though, It could be added to your master/examples folder, and the changes in SonicDiscReader_B.ino could be merged to SonicDiscReader.ino (only parsing out final output was altered. I could then add a readme that explains how the Proc3 sketch works. I could do that if it works for you. I don't think it would confuse someone who doesn't know processing into not using SonicRiscReader if I explain it well.

I'm fairly new to github. if I do this in my folder and then recommit, will it let you selectively pull the processing3 sketch without the schematics/dxf files? (if we go this route)

New Proposal: 
* Keep flat disc eagle/dxf files separate (in my fork)
* SonicDiscReader_B.ino merged into SonicDiscReader.ino
* SonDisk_Proc3 added to /examples folder

platisd commented:
> For the processing3 test script though, It could be added to your master/examples folder, and the changes in SonicDiscReader_B.ino could be merged to SonicDiscReader.ino (only parsing out final output was altered. I could then add a readme that explains how the Proc3 sketch works. I could do that if it works for you. I don't think it would confuse someone who doesn't know processing into not using SonicRiscReader if I explain it well.

> New Proposal:
> * Keep flat disc eagle/dxf files separate (in my fork)
> * SonicDiscReader_B.ino merged into SonicDiscReader.ino
> * SonDisk_Proc3 added to /examples folder

I don't mind adding/merging the processing and the arduino sketches as long as that makes sense and it is well documented. :+1:  Since I don't have a very clear picture on the changes you want to introduce, as they appear as completely new files and thus I would have to manually compare them, I suggest the following:
* Your master (or whichever branch you choose to merge with this `master`) has to contain the end-result of both repositories after the merge is complete. It is not possible to only "pick" specific files from this pull request, unless I copy them manually which would defeat the purpose of the pull request. :stuck_out_tongue: 
* Since you probably want to keep your stuff, I would suggest that you branch out now so to keep a branch with your repo-specific version. So you can do something like `git checkout -b sonicring_fork`.
* Then move/rename the files you want to merge with mine and remove the rest.
* Finally, I don't remember if you have to press a button somewhere so to "update" this pull request or just adding new commits on your `master` will make them appear here. I think they will automatically appear here in which case you don't need to do something more.

richardFirth commented:
This makes sense. I'm still fairly new to git, so i'd like to do everything through git as part of learning.
I'll have a chance to update the stuff probably late this week, so it'll be a few days before i have it in place.
 I'll leave another comment here when I've got it to the state to be pulled.

platisd commented:
Cool, looking forward to it!

platisd commented on examples/README.md:
```diff
@@ -0,0 +1,26 @@
+# Sonic Disc Reader
```
I don't find a central `README.md` as sustainable since this text file will be growing the more examples are added therefore this will make each example's documentation less discoverable. My suggestion is to follow the previous scheme by having a `README.md` in each example folder.

richardFirth replied:
This makes sense. I've replaced the original readme files and moved mine into the processing3 sketch folder.

platisd commented on examples/README.md:
```diff
@@ -0,0 +1,26 @@
+* Read data
+* Filter data based on variance
+
+# SonDisk_Proc3
```
Choose a different, more descriptive, name maybe? Also use Dis**c** instead of Dis**k**. :smile:

richardFirth replied:
renamed to DesktopDisplay_SonicDisc
renamed SonicRing_1.pde to SonicDisc_1.pde

platisd commented on examples/README.md:
```diff
@@ -0,0 +1,26 @@
+and then open the processing sketch. Use the number keys to select the COM port for the arduino, and the output data from SonicDiskReader will be
+displayed in a ring.
+
+![Screenshot](Proc3.png)  
```
Please upload the image on [imgur](imgur.com). :+1:

richardFirth replied:
I got rid of the image since I don't have an imgur account and it didn't seem to add that much value.

platisd commented on examples/SonDisk_Proc3/SerialConnectorInput_D.pde:
```diff
@@ -0,0 +1,34 @@
+     sonic6 = int(theValues[7]);
+     sonic7 = int(theValues[8]);
+   }
+return true;
```
Can you run some auto-indentation over the file?

richardFirth replied:
used CTRL-T on each tab

platisd commented on examples/SonDisk_Proc3/SerialConnector_4.pde:
```diff
@@ -0,0 +1,173 @@
+}
+
+
+*/
```
Please remove commented out code. If it is an example on how to use an API or something similar, please designate it so.

richardFirth replied:
code removed

platisd commented on examples/SonDisk_Proc3/SerialConnector_4.pde:
```diff
@@ -0,0 +1,173 @@
+        }
+        ConnectionStillValid();
+        
+          if (portRef==null){  fill(255,0,0);} else {  fill(0,255,255);}
```
Please indent the code.

platisd commented on examples/SonDisk_Proc3/SerialConnector_4.pde:
```diff
@@ -0,0 +1,173 @@
+        isConnected = true;
+        connectionPort = Serial.list()[portIndex];        
+      }catch(Exception e){
+      System.err.println("counldn't connect");
```
Indent code! :)

richardFirth replied:
auto indent run on all .pde files

richardFirth commented:
Hey, I've updated my version to the changes discussed.
* SonicDiscReader_B.ino merged into SonicDiscReader.ino
* SonDisk_Proc3 added to /examples folder
* in addition, i combined the readme's in SonicDiscReader and smartcar into one readme in the Examples folder. This was so i could have my documentation of sonDisk_Proc3 next to that of SonicDiscReader.ino

platisd commented on examples/SonDisk_Proc3/SerialConnector_4.pde:
```diff
@@ -0,0 +1,173 @@
+  
+  
+  
+  
```
Remove extra newlines.

platisd commented on examples/SonDisk_Proc3/SerialConnector_4.pde:
```diff
@@ -0,0 +1,173 @@
+        if (aKey == 'b') changeBaud();
+    }
+
+  
```
Remove extra newlines

richardFirth replied:
removed extra newlines

platisd commented on examples/README.md:
```diff
@@ -0,0 +1,26 @@
+# SonDisk_Proc3
+This sketch provides a visual interface for the data produced by SonicDiskReader. Connect an arduino with a SonicDisc to the computer,
+and then open the processing sketch. Use the number keys to select the COM port for the arduino, and the output data from SonicDiskReader will be
+displayed in a ring.
```
You can have all of these on the same line.

platisd replied:
Additionally, it would be great if you can describe what is the purpose for each of the `processing` files, either in the readme or within each file.

richardFirth replied:
updated documentation within the processing files.

platisd commented on examples/SonDisk_Proc3/SonDisk_Proc3.pde:
```diff
@@ -0,0 +1,55 @@
+TextDisplayBox tb6;
+TextDisplayBox tb7;
+TextDisplayBox tb8;
+*/
```
Please remove commented out code.

richardFirth replied:
removed commented out code

platisd commented on examples/SonDisk_Proc3/SonDisk_Proc3.pde:
```diff
@@ -0,0 +1,55 @@
+   tb6.drawBox(sonic5);
+   tb7.drawBox(sonic6);
+   tb8.drawBox(sonic7);
+   */
```
Please remove commented out code.

richardFirth replied:
removed commented out code

platisd commented on examples/SonDisk_Proc3/SonicRing_1.pde:
```diff
@@ -0,0 +1,73 @@
+  
+  
+  
+  
```
Remove newlines

richardFirth replied:
removed newlines

richardFirth commented:
Hey, thanks for the detailed feedback, I appreciate the time taken to help me document my stuff in a way others will understand better. 

I've updated made the recommended changes. The reasoning makes sense to me, I just never noticed it myself because I am used to my own code and didn't notice it.

The processing3 code is spread out between separate code files because I use a separate program to sync them to the latest version (that I haven't documented/properly finished yet) https://github.com/richardFirth/CodeGeneratingMacros/blob/master - ProcCodeManager
This program sync's files with the same name to the latest version among versions. Hence if I change the "SerialConnector_4" in one sketch, I can use the macro to copypaste that version to all the other versions to include the update. That's why there are functions that exist that aren't called in this sketch. 

The processing3 <-> arduino link of parsing code was also somewhat auto generated by some vba code (SerialConnector_14Jan17.xlsm) that spits out the variable parsing function and arduino sending function, and has some autogenerated objects that weren't even used in this sketch. I have removed them from this file so there now only the used files.

platisd commented on examples/DesktopDisplay_SonicDisc/SerialConnector_4.pde:
```diff
@@ -0,0 +1,175 @@
+      connectionPort = Serial.list()[portIndex];
+    }
+    catch(Exception e) {
+      System.err.println("counldn't connect");
```
*couldn't :smile:

richardFirth replied:
fixed.

platisd commented on examples/DesktopDisplay_SonicDisc/README.md:
```diff
@@ -0,0 +1,6 @@
+
+
```
Remove newlines

richardFirth replied:
removed newlines

platisd commented on examples/DesktopDisplay_SonicDisc/DesktopDisplay_SonicDisc.pde:
```diff
@@ -0,0 +1,30 @@
+SerialConnector sc; // Serial connection object
+SonicDisc SD;// SonicDisc object
+
+int sonic0 = 50;int sonic1 = 50;int sonic2 = 50;int sonic3 = 50;int sonic4 = 50;int sonic5 = 50;int sonic6 = 50;int sonic7 = 50; // declare int values to which data will be posted
```
Declare these variables on separate lines maybe? :smiley:

richardFirth replied:
split onto separate lines

platisd commented on examples/DesktopDisplay_SonicDisc/README.md:
```diff
@@ -0,0 +1,6 @@
+
+
+# DesktopDisplay_SonicDisc
```
To someone (like me) that has not used Processing before, is it straight forward how to run this? Are there any necessary dependencies or would I just need to download Processing open the project and click run?

richardFirth replied:
good point, I've added a link to the processing webpage to download it

platisd replied:
Sounds exciting, but where is the commit? :smiley: 

Looking forward to merge this!

platisd commented on README.md:
```diff
@@ -97,3 +97,6 @@ Below you can find the necessary components to build a SonicDisc yourself:
 * 1 x 6-pin stackable header
+
+## Derivative works
+Version that uses Arduino Pro mini and has sensors on single plane -> https://github.com/richardFirth/sonicdisc
```
Please use markdown for links! For example my suggestion is a bullet point like this:
* [Arduino Pro Mini version with sensors on single plane](https://github.com/richardFirth/sonicdisc) by [richardFirth](https://github.com/richardFirth)

richardFirth commented:
link updated.
I like the MD links better, I can see it makes it cleaner. I might update my repositorys to use them at some point.
